src/main/java/com/exasol/sql/expression/function/exasol/ExasolFunction.java

```diff
@@ -31,14 +32,18 @@ public static ExasolFunction of(final FunctionName functionName) {
     /**
      * Create a new {@link ExasolFunction} instance.
      *
-     * @param functionName name of the function
+     * @param functionName     name of the function
      * @param valueExpressions zero or more value expressions
      * @return new {@link ExasolFunction}
      */
     public static ExasolFunction of(final FunctionName functionName, final ValueExpression... valueExpressions) {
         return new ExasolFunction(functionName, Arrays.asList(valueExpressions));
     }
 
+    public static ExasolFunction castScalarFunction(final ValueExpression valueExpression, final DataType dataType) {
```

Comment:
the name is strange. Example:
`castScalarFunction(NullLiteral.nullLiteral(), DECIMAL)`
Since the literal is not a scalar function the name does not fit.